### PR TITLE
Test Stability Enhancements

### PR DESCRIPTION
**Pull request by KingOfBrian**
Hey @dostrander and @ahtierney , A few stability enhancements I thought you might be interested in. I'm curious to see if this helps your automated tests be more reliable. There may be a small change needed to the unit tests.

- Move back to travis. Easier testing of 7.3 + 8.0
- Expiration had a few corner cases that weren't aggravated well by the unit tests and were causing intermittent failures. 
- Unit tests check both the idle state of the simulated central and the number of dispatch commands
- Disable some asynchronous behavior in the `RZBSimulatedDevice` when not needed.
- Re-write the delay implementation in `RZBSimulatedCallback`. It was confusing.
- Fix a few typo's.

### CODE REVIEW

File: RZMockBluetooth/Simulation/RZBSimulatedCallback.m [hunk 1]
```diff
@@ -56,7 +56,9 @@ - (void)dispatch:(RZBSimulatedCallbackBlock)block
 
     dispatch_group_enter(self.group);
     dispatch_after(dispatch_time(DISPATCH_TIME_NOW, (int64_t)(self.delay * NSEC_PER_SEC)), self.queue, ^{
-        dispatch_group_leave(wself.group);
+        if (wself.group) {
```
Comment (ahtierney) [inline RZMockBluetooth/Simulation/RZBSimulatedCallback.m]: 👍

**Comment by ahtierney** [comment]
@KingOfBrian This looks good to me, when I have some more time I'll look at getting it integrated into Nima. The changes seem to address exactly what I was seeing with the SimulatedDevice as well as making the simulated internals more inspectable via. `idle`.

**Comment by KingOfBrian** [comment]
Sweet -- I'd like to get a release out before the end of the week, so I can move on to some other stuff.
